dm/syncer/syncer.go

```diff
@@ -961,18 +967,16 @@ func (s *Syncer) addJob(job *job) error {
 		})
 	}
 
-	// nolint:ifshort
-	needFlush := s.checkFlush()
 	failpoint.Inject("flushFirstJob", func() {
```

Comment:
actually we still have sync flush, the async flush only applies for those flush happened on every flush time interval. But we still support sync flush used by ddl, heardbeat event. (can be found by search usage of `flushJobs()`)

Reply:
yes, sync flush is still used. 

The original bug is not about sync or async flush, it is caused by the time of triggering a flush: when we trigger a flush because of `CheckGlobalPoint` in the middle of a multi-row event, we may save a location forwarded too much. Now the real code logic of "trigger a flush in the middle of a multi-row event" use async flush, we should test async flush.

Reply:
got it!